contrib/qubes/src/updatevm/fwupd-download-updates.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+FWUPD_UPDATEVM_DIR=/home/user/.cache/fwupd
+
+echo "Running fwupd download script..."
+
+CLEAN=
+CHECK_ONLY=
+METADATA=
+SHASUM=
+UPDATE=
+URL=
+FW_NAME=
+
+while [ -n "$1" ]; do
+    case $1 in
+        --CHECK_ONLY)
+            CHECK_ONLY=1
+            ;;
+        --clean)
+            CLEAN=1
+            ;;
+        --metadata)
+            METADATA=1
+            ;;
+        --url=*)
+            URL=${1#--url=}
+            FW_NAME=${1#--url=https://fwupd.org/downloads/}
+            ;;
+        --sha=*)
+            UPDATE=1
+            SHASUM=${1#--sha=}
+            ;;
+        -*)
+            echo "Command $1 unknown. exiting..."
+            exit 1
+            ;;
+        *)
+            echo "Command $1 unknown. exiting..."
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ ! -d $FWUPD_UPDATEVM_DIR ]; then
+    echo "fwupd updates dir does not exist: $FWUPD_UPDATEVM_DIR" >&2
+    exit 1
+fi
+
+if [ "$CHECK_ONLY" == "1" ]; then
+    echo "Check only mode."
+fi
+
+if [ "$CLEAN" == "1" ]; then
+    echo "Cleaning cache."
+    rm -rf $FWUPD_UPDATEVM_DIR/metadata/*
+    rm -rf $FWUPD_UPDATEVM_DIR/updates/*
+fi
+
+if [[ "$URL" == *"--and--"* ]]; then
+    URL=${URL//--and--/&}
+    FW_NAME="untrusted.cab"
+fi
+
+if [[ "$URL" == *"%20"* ]]; then
+    FW_NAME="untrusted.cab"
+fi
+
+if [[ "$URL" == *"--or--"* ]]; then
+    URL=${URL//--or--/|}
+    FW_NAME="untrusted.cab"
+fi
+
+if [[ "$URL" == *"--hash--"* ]]; then
+    URL=${URL//--hash--/#}
+    FW_NAME="untrusted.cab"
+fi
+
+if [ "$METADATA" == "1" ] && [ -z "$URL" ]; then
+    echo "Downloading metadata."
+    rm -rf $FWUPD_UPDATEVM_DIR/metadata/*
+    wget -P $FWUPD_UPDATEVM_DIR/metadata \
+        https://fwupd.org/downloads/firmware.xml.gz
+    wget -P $FWUPD_UPDATEVM_DIR/metadata \
+        https://fwupd.org/downloads/firmware.xml.gz.jcat
+    wget -P $FWUPD_UPDATEVM_DIR/metadata \
+        https://fwupd.org/downloads/firmware.xml.gz.asc
+    gpg --verify $FWUPD_UPDATEVM_DIR/metadata/firmware.xml.gz.asc \
+        $FWUPD_UPDATEVM_DIR/metadata/firmware.xml.gz
+    if [ ! $? -eq 0 ]; then
+        echo "Signature did NOT match. Exiting..."
+        exit 1
+    fi
+fi
+
+if [ "$METADATA" == "1" ] && [ -n "$URL" ]; then
+    echo "Downloading metadata."
+    rm -rf $FWUPD_UPDATEVM_DIR/metadata/*
+    wget -P $FWUPD_UPDATEVM_DIR/metadata $URL
+    wget -P $FWUPD_UPDATEVM_DIR/metadata $URL.jcat
+    wget -P $FWUPD_UPDATEVM_DIR/metadata $URL.asc
```

Comment:
both?

Reply:
We need both files to cover fwupd versions from 0.9.5 to the current one.

Reply:
Why support such old fwupd versions?

Reply:
Ferdora 25 is base for dom0 in Qubes R4.0 (https://www.qubes-os.org/doc/supported-versions/) and it provides fwupd 0.9.5.

Reply:
The file doesn't exist. It's replaced with the python script. There is no need to download the GPG signature, thus I left only `jcat` file. The wrapper works correctly with the current libjcat master and the changes connected to the https://github.com/hughsie/libjcat/issues/48 and https://github.com/hughsie/libjcat/issues/49 need to be available on Fedora 32.

Reply:
libjcat update just went stable for f32: https://bodhi.fedoraproject.org/updates/FEDORA-2021-a83da4c566